apps/phaseofmoon/phase_of_moon.star

```diff
@@ -0,0 +1,116 @@
+  ]
+
+# Phase of the moon data. 
+# Dates in phase changes give 2 days for key phases and lengthen others to match. Removes over-display of key phases.
+# Idea, data and calculation from https://minkukel.com/en/various/calculating-moon-phase/
+NUM_PHASES = 8
+MOON_PHASES = ["New Moon", "Waxing Crescent", "First Quarter", "Waxing Gibbous", "Full Moon", "Waning Gibbous", "Last Quarter", "Waning Crescent", "New Moon", ]
+PHASE_CHANGES = [0, 1, 6.38264692644, 8.38264692644, 13.76529385288, 15.76529385288, 21.14794077932, 23.14794077932, 28.53058770576, 29.53058770576,]
+
+def main(config):
+    # Get latitude from location
+    location = config.get("location")
+    loc = json.decode(location) if location else DEFAULT_LOCATION
+    lat = float(loc.get("lat"))
+    # use latitude to work out which hemisphere we're in
+    hemisphere = 1 if lat >= 0 else 0
+
+    # Get the current time
+    now = time.now().in_location(DEFAULT_TIMEZONE).unix
+
+    # Get the current fraction of the moon cycle
+    currentfrac = math.mod(now - FIRSTMOON, LUNARSECONDS) / LUNARSECONDS
+    # Calculate current day of the cycle from there
+    currentday = currentfrac * LUNARDAYS
+
+
+    for x in range(0, NUM_PHASES):
+        if currentday > PHASE_CHANGES[x] and currentday <= PHASE_CHANGES[x + 1] :
+            moonPhase = MOON_PHASES[x]
+            phaseImage = PHASE_IMAGES[x]
+            if hemisphere == 0 :
+                phaseImage = PHASE_IMAGES[NUM_PHASES - x]
+        
+
+    return render.Root(
+        child = render.Box(
+            render.Row(
+                expanded=True,
+                main_align="start",
+                cross_align="center",
+                children = [
+                    render.Image(src=base64.decode(phaseImage)),
+                    render.Padding(
+                        pad=(1,0,0,0),
+                        child =  render.WrappedText(moonPhase),
+                    )
+                ]
+            )
+        )
+    )
+    
```

Comment:
You will need this schema field to ensure we populate the `location` config object in `location = config.get("location")`:

```suggestion

def get_schema():
    return schema.Schema(
        version = "1",
        fields = [
            schema.Location(
                id = "location",
                name = "Location",
                desc = "Location for which to display the moon phase for.",
                icon = "place",
            ),
        ],
    )
```

Reply:
How did I forget that? I even had the module added! On its way.